### PR TITLE
Add parallel decoding of media file in a collection

### PR DESCRIPTION
**Pull request by padenot**
This is no-op because nothing uses it, but does printouts on `stderr` so that tests still pass.

`audrey` is used for flac and wav and flac decoding. It panics a lot on my collection, we'll sort this out later.
`minimp3` is used for mp3 decoding, this works well.

This also includes a small fix so that `clippy` is happier.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,6 +7,9 @@ authors = ["Raphaël Lustin <raphael@lustin.fr>"]
 clap = { version = "2.33.0", features = ["yaml"] }
 quick-xml = { version = "0.17", features = ["serialize"] }
 serde = { version = "1.0", features = ["derive"] }
+minimp3 = "0.3"
+audrey = "0.2"
+rayon = "1.3"
```
Comment (rlustin) [inline Cargo.toml]: Could you sort dependencies by alphabetical order?

File: src/main.rs [hunk 2]
```diff
@@ -1,14 +1,132 @@
...
+use std::ffi::OsStr;
+use std::fs::File;
+use std::path::Path;
 use traktor::parse_traktor_collection;
 
 mod error;
 mod models;
 mod traktor;
 
+struct DecodedFile {
```
Comment (rlustin) [inline src/main.rs]: Could you extract the model into a module?

File: src/main.rs [hunk 3]
```diff
@@ -1,14 +1,132 @@
...
+            }
+            Err(e) => {
+                eprintln!("{}", e);
+            }
+        }
+        // analyse
+        // out integrated lufs
+    });
+}
+
```
Comment (rlustin) [inline src/main.rs]: Could you extract decoding into its dedicated module?

File: src/main.rs [hunk 4]
```diff
@@ -1,14 +1,132 @@
...
+        Err("file not found")
+    }
+}
+
+fn handle_minimp3(path: &str) -> Result<DecodedFile, &str> {
+    match File::open(path) {
+        Ok(f) => {
+            let mut decoder = Decoder::new(f);
+            let mut pcm_data = Vec::<f32>::new();
+            let mut sr: i32 = 0;
```
Comment (rlustin) [inline src/main.rs]: `sr` -> `sample_rate`?

File: src/main.rs [hunk 5]
```diff
@@ -1,14 +1,132 @@
...
+            let mut ch: u32 = 0;
```
Comment (rlustin) [inline src/main.rs]: `ch` -> `channel_count`?

File: src/main.rs [hunk 6]
```diff
@@ -1,14 +1,132 @@
...
+    data: Vec<f32>, // interleaved
+}
+
+fn handle_audrey(path: &str) -> Result<DecodedFile, &str> {
+    let maybe_file = audrey::read::open(&path);
+    if let Ok(mut file) = maybe_file {
+        let desc = file.description();
+        let data: Vec<f32> = file.samples().map(Result::unwrap).collect::<Vec<_>>();
+
+        Ok(DecodedFile {
```
Comment (rlustin) [inline src/main.rs]: Could you add a new line after parenthesis opening?

File: src/main.rs [hunk 7]
```diff
@@ -1,14 +1,132 @@
...
+fn handle_minimp3(path: &str) -> Result<DecodedFile, &str> {
+    match File::open(path) {
+        Ok(f) => {
+            let mut decoder = Decoder::new(f);
+            let mut pcm_data = Vec::<f32>::new();
+            let mut sr: i32 = 0;
+            let mut ch: u32 = 0;
+            let rv = loop {
+                match decoder.next_frame() {
+                    Ok(Frame {
```
Comment (rlustin) [inline src/main.rs]: Could you add a new line after parenthesis opening?

File: src/main.rs [hunk 8]
```diff
@@ -1,14 +1,132 @@
...
+                        });
+                    }
+                    Err(Error::Eof) => break Ok(pcm_data),
+                    Err(_) => {
+                        break Err("mp3 corrupted");
+                    }
+                }
+            };
+            match rv {
+                Ok(data) => Ok(DecodedFile {
```
Comment (rlustin) [inline src/main.rs]: Could you add a new line after parenthesis opening?

**Review by rlustin — CHANGES_REQUESTED** [review]

**Comment by padenot** [comment]
PTAL

**Review by rlustin — APPROVED** [review]

**Pull request merged**